### PR TITLE
OADP-4137: updated for velero 1.14 (#1398)

### PR DESCRIPTION
**Pull request by sseago**
* updated for velero 1.14

go.mod changes for Velero 1.14 (including golang 1.22) velero CRD changes for 1.14
some oadp operator code changes to cope with controller-runtime upgrade to 0.17.2

* fix tests

* use go 1.22.2

* update tags to 1.4

* update to controller-gen 0.14.0

* fix deploy-olm for 4.16

* make bundle diff for operator-sdk 1.34

* leaderelection: Warning->Warningf

* run go fmt/generate

* Remove createdAt only change

* fix [OADP-3775](https://issues.redhat.com//browse/OADP-3775): Investigate and make OADP Operator ready for CSI plugin as an internal Velero plugin

* remove irrelevant comments

* remove line number from leaderelection lease check

* don't check for CSI plugin initcontainer

* filter out repo-maint job pods from Velero pod query

* account for restore Finalizing in e2e

---------

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -433,7 +433,7 @@ catalog-push: ## Push a catalog image.
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
 # A valid Git branch from https://github.com/openshift/oadp-operator
-PREVIOUS_CHANNEL ?= oadp-1.2
+PREVIOUS_CHANNEL ?= oadp-1.3
```
Comment (mateusoliveira43) [inline Makefile:436]: for master, should be 1.4 (highest release), but can be follow up PR :smile:

Comment (sseago) [reply Makefile:436]: @mateusoliveira43 Probably doesn't matter much here, but I'd say 1.4 would be appropriate once 1.4.0 is officially out.

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
`make test` failed to me

I assume we need this in https://github.com/openshift/oadp-operator/pull/1279

File: bundle/manifests/oadp.openshift.io_dataprotectionapplications.yaml [hunk 2]
```diff
@@ -1069,5 +1392,5 @@ status:
   acceptedNames:
     kind: ""
     plural: ""
-  conditions: []
-  storedVersions: []
+  conditions: null
```
Comment (shubham-pampattiwar) [inline bundle/manifests/oadp.openshift.io_dataprotectionapplications.yaml:1395]: I am not sure how this change would affect us: `[]` -> `null` , maybe its harmless

Comment (mateusoliveira43) [reply bundle/manifests/oadp.openshift.io_dataprotectionapplications.yaml:1395]: I think its controller gen default that changed

but we also did this change on the other PR https://github.com/openshift/oadp-operator/pull/1398/files#r1617718698

Comment (sseago) [reply bundle/manifests/oadp.openshift.io_dataprotectionapplications.yaml:1395]: @shubham-pampattiwar @mateusoliveira43 Yes, controller-gen upgrade changed the way some of this is rendered. This matches oadp-1.4.

**Review by mateusoliveira43 — APPROVED** [review]
CI failures I spot were flakes (restore succeeds, but application not working)

**Review by shubham-pampattiwar — APPROVED** [review]
/lgtm

**Review by kaovilai — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
/override ci/prow/4.14-e2e-test-kubevirt-aws
/override ci/prow/4.13-e2e-test-aws
flake: restore succeeds, but application not working

**Comment by weshayutin** [comment]
nice work all :)  Have a great weekend 0/

**Pull request merged**